backend/routes/items.js

```diff
@@ -125,4 +125,37 @@ router.post("/addItem", token_required, async (req, res, next) => {
   });
 });
 
+router.put("/editItem", token_required, async (req, res, next) => {
+  // authentication
+  try {
+    // req.user is the user id fetched from Middleware
+    const user = await User.findById(req.user.id);
+  } catch (e) {
+    console.log(e);
+    return res.send({ message: "Error in Fetching user" }, 401);
+  }
+
+  try {
+    const { itemName, batchId, weight } = req.body;
+    const outDate = toUTCMidnight(new Date(req.body.outDate));
+    let batch = await Batch.findOne({ itemName, batchId });
+    
+    if (outDate < batch.inDate) {
+      return res.status(400).json({
+        fieldName: "outDate",
+        message: "Outdate is in the past",
+      });
+    }
+    if (weight > batch.poundsTotal) {
```

Comment:
I'm a little confused why we need to check for this. Is this saying that if there are only 50 pounds left to ship out, the user cannot edit the weight to be 80, or am I understanding this incorrectly? The user should be able to edit the number of pounds to any positive number, even if it is larger than the current total number of pounds to ship out.

Reply:
I was thinking that the poundsTotal is how much supply the user has of a batch of item. So if a user adds a batch of 50 pounds, the poundsTotal would be set to 50, and if they edit the batch whenever they send out a certain number of pounds, the poundsRemaining would be set to that weight they put. So the weight would be how much they still have to send out. Correct me if I'm wrong!

Reply:
I think you are correct in that is the main usage of the edit button, however I was also thinking about the case the user accidentally slips up. For example, maybe they entered a batch of 50 pounds but they actually meant a batch of 500 pounds (missed a 0). I think we need to be able to allow for changing the total number of pounds for errors like that.

Reply:
Fixed!